packages/network-controller/src/NetworkController.ts

```diff
@@ -1354,19 +1352,20 @@ export class NetworkController extends BaseController<
    * removed in a future release
    */
   async setProviderType(type: InfuraNetworkType) {
-    assert.notStrictEqual(
-      type,
-      NetworkType.rpc,
-      // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      `NetworkController - cannot call "setProviderType" with type "${NetworkType.rpc}". Use "setActiveNetwork"`,
-    );
-    assert.ok(
-      isInfuraNetworkType(type),
-      // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      `Unknown Infura provider type "${type}".`,
-    );
+    if ((type as unknown) === NetworkType.rpc) {
+      throw new Error(
+        // TODO: Either fix this lint violation or explain why it's necessary to ignore.
```

Comment:
Nit: This is another false positive, we can update this TODO to reflect that.

Reply:
Updated in [557c1a4](https://github.com/MetaMask/core/pull/3672/commits/557c1a4a31e9e33e9fe8f2a7c063901fb1798856).